### PR TITLE
Upgrading Go to 1.19

### PR DESCRIPTION
**Pull request by khrm**
Our CI uses go1.19. Also, 1.18 is EOL.

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#tests) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#docs) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commits)
- [x] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Review by khrm — COMMENTED** [review]
/kind misc

File: third_party/github.com/hashicorp/go-retryablehttp/client.go [hunk 1]
```diff
@@ -601,7 +602,7 @@ func (c *Client) Do(req *Request) (*http.Response, error) {
 			if c, ok := body.(io.ReadCloser); ok {
 				req.Body = c
 			} else {
-				req.Body = io.NopCloser(body)
+				req.Body = ioutil.NopCloser(body)
```
Comment (khrm) [inline third_party/github.com/hashicorp/go-retryablehttp/client.go:605]: An update to go1.19 shouldn't change this. Any idea why this happened? Also, why do we need go code here?

Comment (dibyom) [reply third_party/github.com/hashicorp/go-retryablehttp/client.go:605]: (the go code is due to the license terms of the particular library I think)

**Review by dibyom — APPROVED** [review]
/lgtm

**Pull request merged**